api/client/client.go

```diff
@@ -62,14 +59,24 @@ type Client struct {
 	closedFlag int32
 }
 
-// NewClient returns a new auth client that uses mutual TLS authentication and
-// connects to the remote server using the Dialer or Addrs given in Config.
-func NewClient(cfg Config) (*Client, error) {
+// New creates a new API client with a connection to a Teleport server. The connection is
+// formed using the Dialer and CredentialsProviders given in config. If Addrs are provided, a
+// basic Dialer will be created. If Credentials are provided, a basic CredentialsProvider will
+// will be added to the list of CredentialsProviders.
+func New(cfg Config) (*Client, error) {
 	if err := cfg.CheckAndSetDefaults(); err != nil {
 		return nil, trace.Wrap(err)
 	}
 
 	c := &Client{c: cfg}
+	if err := c.connect(cfg.NoPingCheck); err != nil {
```

Comment:
No need to pass this into connect. If connect succeeds and this is set, just invoke Ping afterwards.

Reply:
The ping is used to test the connection and credentials after they're loaded, so it needs to be done within the for loop in `connect`. I should update it to read the client's config rather than passing the variable in though.